kolibri/plugins/learn/assets/src/my_downloads/views/DownloadsList/index.vue

```diff
@@ -128,25 +131,70 @@
     setup() {
       const { genExternalContentURLBackLinkCurrentPage } = useContentLink();
       const { getLearningActivityIcon } = useLearningActivities();
-
+      const { downloadRequestMap, availableSpace } = useDownloadRequests();
+      const totalPageNumber = ref(0);
+      const store = getCurrentInstance().proxy.$store;
+      const query = computed(() => get(route).query);
+      const route = computed(() => store.state.route);
+      const sort = computed(() => query.value.sort);
+      const pageSizeNumber = computed(() => Number(query.value.page_size || 25));
+      const activityType = computed(() => query.value.activity || 'all');
+      const downloads = ref([]);
+      const sortedFilteredDownloads = () => {
+        let downloadsToDisplay = [];
+        if (downloadRequestMap) {
+          for (const [, value] of Object.entries(downloadRequestMap.value)) {
+            downloadsToDisplay.push(value);
+          }
+          if (sort) {
+            switch (sort.value) {
+              case 'newest':
+                downloadsToDisplay.sort(
+                  (a, b) => new Date(b.requested_at) - new Date(a.requested_at)
+                );
+                break;
+              case 'oldest':
+                downloadsToDisplay.sort(
+                  (a, b) => new Date(a.requested_at) - new Date(b.requested_at)
+                );
+                break;
+              case 'smallest':
+                downloadsToDisplay.sort((a, b) => a.metadata.file_size - b.metadata.file_size);
+                break;
+              case 'largest':
+                downloadsToDisplay.sort((a, b) => b.metadata.file_size - a.metadata.file_size);
+                break;
+              default:
+                // If no valid sort option provided, return unsorted array
+                break;
+            }
+          }
+          if (activityType) {
+            if (activityType.value !== 'all') {
+              downloadsToDisplay = downloadsToDisplay.filter(download =>
+                download.metadata.learning_activities.includes(activityType.value)
+              );
+            }
+          }
+        }
+        set(totalPageNumber, Math.ceil(downloadsToDisplay.length / pageSizeNumber.value));
+        set(downloads, downloadsToDisplay);
+      };
+      sortedFilteredDownloads();
+      watch(route, () => {
```

Comment:
Rather than doing an explicit watch here, we could just initialize `downloads` as a computed value that returns the results of the `sortedFilteredDownloads` function.

`totalPageNumber` can then also be a computed value based on the `downloads` computed value.